localstack/services/stepfunctions/asl/component/state/state_execution/state_map/item_reader/reader_config/max_items_decl.py

```diff
@@ -48,7 +65,45 @@ class MaxItemsPath(MaxItemsDecl):
     def __init__(self, path: str):
         self.path: Final[str] = path
 
+    def _validate_value(self, env: Environment, value: int) -> None:
+        if not isinstance(value, int):
+            # TODO: Note, this error appears to be validated at a earlier stage in AWS Step Functions, unlike the
```

Comment:
interesting and worth commenting on 👍 . Just out of curiosity, how would we approach fixing this in the future?
* Does it require figuring out where it happens?
* Would we need to introduce extra validation steps?

Reply:
This needs more validation tests first to accurately pin point when, and possibly why, the two validations are carried out at different depths. In terms of changes, it seems quite likely that the evaluation order within the map run will need some adjusting, ad some validation errors about MapItemPath are raised before MapStarted events, whilst some are raised after. This change would not be a trivial one, that is why I decided to postpone it to a future PR. As you identified, this PR mainly wanted to address the jsonpath extraction logic (and the max item evaluation as a result).

Reply:
Very reasonable 👍 Thanks for the clarification.